### PR TITLE
feat(core): support webpack-manifest-plugin

### PR DESCRIPTION
**Pull request by hardfist**
## Related issue (if exists)
related to #2196, support webpack-manifest-plugin, since we doesn't support custom stage and some community relys on custom stage, so we fallback unsupported stage to supported stage now and gives warning for it.

the forked repo and the changes is here https://github.com/hardfist/rspack-manifest-plugin/pull/1

### <samp>🤖 Generated by Copilot at 8b7f270</samp>

Added support for generating asset manifest files with `webpack-manifest-plugin` in `rspack`. This improves compatibility with some webpack plugins and enables integration with other tools that need to know the output file names. Updated the `plugin-compat` example to demonstrate this feature.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at 8b7f270</samp>

*  Add `webpack-manifest-plugin` dependency to `examples/plugin-compat/package.json` and `pnpm-lock.yaml` ([link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-7c79896436e137329e49b20866aa4502be0fe10e49a15ceb45974993392878c1L21-R22), [link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR415), [link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR425), [link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR22836-R22845))
* Import and configure `WebpackManifestPlugin` in `examples/plugin-compat/rspack.config.js` to generate `rspack-manifest.json` in output directory ([link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-406f4d16e21e23ae6783e825f72f72378a25ff933d3fbc3db1d2029478f9ff96R7), [link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-406f4d16e21e23ae6783e825f72f72378a25ff933d3fbc3db1d2029478f9ff96R59-R61))
* Implement `isOnlyInitial` method on `chunk` object in `packages/rspack/src/compilation.ts` for compatibility with plugins that filter or process initial chunks ([link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-007fbe3e2cbba14328e3a87deaa9b6557a2d94e268cc07795c9a235a5b0575a1L630-R633))
* Validate and normalize `stage` parameter of `processAssets` method in `packages/rspack/src/compilation.ts` for compatibility with plugins that use custom stages for asset processing ([link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-007fbe3e2cbba14328e3a87deaa9b6557a2d94e268cc07795c9a235a5b0575a1R789-R804))
* Add `source-list-map` and `webpack-sources` dependencies to `pnpm-lock.yaml` as transitive dependencies of `webpack-manifest-plugin` ([link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR20474-R20477), [link](https://github.com/web-infra-dev/rspack/pull/3163/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR22854-R22861))

</details>

### CODE REVIEW

File: examples/plugin-compat/rspack.config.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const CopyPlugin = require("copy-webpack-plugin");
 const HtmlPlugin = require("@rspack/plugin-html").default;
 const { StatsWriterPlugin } = require("webpack-stats-plugin");
 const minifyPlugin = require("@rspack/plugin-minify");
+const manifestPlugin = require("rspack-manifest-plugin").WebpackManifestPlugin;
```
Comment (chenjiahan) [inline examples/plugin-compat/rspack.config.js:7]: looks a bit strange

Comment (hardfist) [reply examples/plugin-compat/rspack.config.js:7]: I may make webpack-manifest-plugin one day, but in order to make minimal change so I don't change internal export variable

Comment (chenjiahan) [reply examples/plugin-compat/rspack.config.js:7]: get👌🏻

**Comment by hardfist** [comment]
!canary

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**